src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public DateTimeFormatter getFormatter() {
+        return formatter;
+    }
+
+    public void setFormatter(DateTimeFormatter formatter) {
+        this.formatter = formatter;
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter).plusDays(1);
+        LocalDate[] dates = new LocalDate[data.length];
+        String[] namesOfData = new String[data.length];
+        int[] hours = new int[data.length];
+        int[] rate = new int[data.length];
+        int[] salary = new int[names.length];
+
+        for (int i = 0; i < data.length; i++) {
+            String[] datas = data[i].split(" ");
+            dates[i] = LocalDate.parse(datas[0], formatter);
```

Comment:
![image](https://user-images.githubusercontent.com/12864128/192104483-a8c82e7d-6588-4962-bb11-285a9d5f0c74.png)

Move 0 to the class constant